onnxruntime/core/session/inference_session.cc

```diff
@@ -1220,8 +1224,46 @@ Status AssignNodesToEpsFromHashes(Graph& graph, const fbs::SessionState& fbs_ses
   ORT_RETURN_IF_ERROR(VerifyEachNodeIsAssignedToAnEp(graph, logger));
   return Status::OK();
 }
-#endif  // defined(ENABLE_ORT_FORMAT_LOAD)
+
+Status TransformGraphForOrtFormatModel(onnxruntime::Graph& graph, const ExecutionProviders& providers,
+                                       KernelRegistryManager& kernel_registry_manager,
+                                       SessionState& session_state,
+                                       const experimental::fbs::SessionState& serialized_session_state,
+                                       const logging::Logger& logger) {
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  // nodes are already partitioned, but a custom EP may compile some at runtime.
+  // run the partitioning to allow that to happen.
+  //
+  // We always have the CPU EP, so only need to run this if some other EP is enabled
+  if (providers.NumProviders() > 1) {
+    ORT_RETURN_IF_ERROR(PartitionOrtFormatModel(graph, providers, kernel_registry_manager,
+                                                session_state));
+  }
+
```

Comment:
We could probably do the partitioning in Initialize as the setup for ORT format models will be different and we will always run the partitioning prior to the transforms (vs. an onnx model where those get interlaced so makes sense for partitioning to be in a 'transform' method).